sonic_y_cable/y_cable_base.py

```diff
@@ -0,0 +1,577 @@
+
+    def get_ber_info(self, port):
+        """
+        This API specifically returns the BER (Bit error rate) value for a specfic port.
+        The target could be local side, TOR1, TOR2, NIC etc.
+
+        Args:
+            physical_port:
+                 an Integer, the actual physical port connected to a Y cable
+            target:
+                 an Integer, the target on which to enable the PRBS
+                             EYE_PRBS_TARGET_LOCAL -> local side,
+                             EYE_PRBS_TARGET_TOR1 -> TOR 1
+                             EYE_PRBS_TARGET_TOR2 -> TOR 2
+                             EYE_PRBS_TARGET_NIC -> NIC
+        Returns:
+            a list, with BER values of lane 0 lane 1 lane 2 lane 3 with corresponding index
+        """
+
+        raise NotImplementedError
+
+    def get_vendor(self, port):
+        """
+        This API specifically returns the vendor name of the Y cable for a specfic port.
+
+        Args:
+            physical_port:
+                 an Integer, the actual physical port connected to a Y cable
+        Returns:
+            a string, with vendor name
+        """
+
+        raise NotImplementedError
+
+    def get_part_number(self, port):
+        """
+        This API specifically returns the part number of the Y cable for a specfic port.
+
+        Args:
+            physical_port:
+                 an Integer, the actual physical port connected to a Y cable
+        Returns:
+            a string, with part number
+        """
+
+        raise NotImplementedError
+
+    def get_switch_count(self, port, count_type):
+        """
+        This API specifically returns the switch count to change the Active TOR which has
+        been done manually by the user.
+
+        Args:
+            physical_port:
+                 an Integer, the actual physical port connected to a Y cable
+            count_type:
+                 a string, for getting the count type
+                          "manual" -> manual switch count
+                          "auto" -> automatic switch count
+        Returns:
+            an integer, the number of times manually the Y-cable has been switched
+        """
+
+        raise NotImplementedError
+
+    def get_target_cursor_values(self, port, lane, target):
+        """
+        This API specifically returns the cursor equalization parameters for a target(NIC, TOR1, TOR2).
+        This includes pre one, pre two , main, post one, post two cursor values
+
+        Args:
+            physical_port:
+                 an Integer, the actual physical port connected to a Y cable
+            lane:
+                 an Integer, the lane on which to collect the cursor values
+                             1 -> lane 1,
+                             2 -> lane 2
+                             3 -> lane 3
+                             4 -> lane 4
+            target:
+                 an Integer, the actual target to get the cursor values on
```

Comment:
Maybe rather than explicitly stating "An Integer", we simply state "One of the following predefined constants:". This would also allow us to change the underlying type transparently if desired.

Reply:
fixed